src/components/auth/profile/MyRecurringCampaignsTable.tsx

```diff
@@ -0,0 +1,141 @@
+import { useTranslation } from 'next-i18next'
+import { useMutation } from '@tanstack/react-query'
+import { AlertStore } from 'stores/AlertStore'
+import { endpoints } from 'service/apiEndpoints'
+import { useSession } from 'next-auth/react'
+
+import { DataGrid, GridColDef, GridColumns, GridRenderCellParams } from '@mui/x-data-grid'
+import { IconButton, Tooltip, Box } from '@mui/material'
+import { money } from 'common/util/money'
+
+import CancelPresentationIcon from '@mui/icons-material/CancelPresentation'
+
+import { useGetUserRecurringDonations } from 'common/hooks/recurringDonation'
+
+import Link from 'components/common/Link'
+import { apiClient } from 'service/apiClient'
+import { authConfig } from 'service/restRequests'
+
+export default function MyRecurringCampaignsTable() {
+  const { t } = useTranslation()
+  const { data: recurringDonations } = useGetUserRecurringDonations()
+  const { data: session } = useSession()
+
+  const cancelMutation = useMutation({
+    mutationFn: async (id: string) => {
+      //TODO: this '/api/v1' does not look right here
+      apiClient.get(
+        endpoints.recurringDonation.cancelRecurringDonation(id).url,
+        authConfig(session?.accessToken),
+      )
+    },
+    onError: (err) => AlertStore.show(t('common:alerts.error') + err, 'error'),
+    onSuccess: () => {
+      AlertStore.show(t('recurring-donation:alerts.cancel'), 'success')
+    },
+  })
+
+  const cancelRecurringDonation = (id: string) => {
+    if (confirm(t('recurring-donation:alerts.cancel-confirm'))) {
```

Comment:
I think we should not use the native `confirm` alert since it looks bad and we should confirm with a MUI Confirm Dialog, that might be for another issue though. We should probably have a reusable confirm dialog component. We might even mimic how this browser api works and create it a function that returns boolean!

Reply:
Fixed.
It turned out we already have such a dialog.